### ISSUE TITLE
Upgrade from 8.2.3 to 8.2.5 gets stuck

### ISSUE THREAD

**Issue by SpamReceiver** [owncloud/core#24715]
See also https://forum.owncloud.org/viewtopic.php?f=36&t=36867&p=116843#p116843

Two days ago my private Owncloud server notified me of an available update to version 8.2.5. As usual I chose to update my server via the updater app on the admin page.

Unfortunately this process gets stuck in this step:
`Aktualisiere ownCloud auf Version 8.2.5. Dies könnte eine Weile dauern.

Update vorbereiten
Log-Level auf Debug gesetzt - aktuelles Level: "Debug"
Wartungsmodus eingeschaltet
Checking whether the database schema can be updated (this can take a long time depending on the database size)
`

Since then I have tried the update several times -- switching maintenance mode off in config.php allows me to restart the update process. Every time no luck: no more progress after "Checking schema...".

Linux top utility shows that mysql is running with high load all the time. This matches the hanging step... an infinite database query?

One update try I kept running for appr. 24 hours with no change in mysql CPU load.

Maybe this issue is related to https://github.com/owncloud/core/issues/24651.  But my scenario is related to the upgrade rather than to a running system.

I am not sure what is meant by "shares" in the bug's comments, but it could be related. I have included directories from localhost (not under control of Owncloud) in Owncloud's files tree which contain ten-thousands of photos and music files.

Something else I found:
I killed the (only one) mysql process while the update was running from the web page (rather than via occ).
The following text was printed on the web page:

`Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'INSERT INTO "oc_oc_filecache_lwp0sijmw3it5" SELECT * FROM "oc_filecache"': SQLSTATE[HY000]: General error: 2006 MySQL server has gone away
`

Server configuration
Machine: QNAP TS-219P II
Operating system: Linux 3.4.6 armv5tel
Web server: Apache/2.2.31
Database: MySQL 5.1.73
PHP version: PHP 5.5.27
ownCloud version (see ownCloud admin page): 8.2.3
Updated from an older ownCloud or fresh install: updated
ownCloud log (data/owncloud.log):

Special configuration (external storage, external authentication, reverse proxy, server-side-encryption): none

```
Login as admin user into your ownCloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.

I can't login... the update process is stuck...
```

**List of activated apps:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your ownCloud installation folder
```

**The content of config/config.php:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your ownCloud installation folder

or 

Insert your config.php content here
(Without the database password, passwordsalt and secret)
```

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

#### LDAP configuration (delete this part if not used)

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your ownCloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```

### Client configuration
**Browser:**

**Operating system:**

#### Web server error log
```
Insert your webserver log here
```

#### ownCloud log (data/owncloud.log)
```
Insert your ownCloud log here
```

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log 
c) ...
```

**Comment by nickvergessen** [comment]
What's the size of your installation?
Are you updating via the web UI or via the console?

If you have console access try to update with `./occ upgrade --skip-migration-test`

**Comment by SpamReceiver** [comment]
How do you measure the installation size? 

I have started the update via the updater app (please see the very top of my report). This left me on the web page with the progress report.

After this has not completed after a reasonable time I reloaded the web page and landed on the web page "Maintenance mode is active". I disabled the maintenance mode by editing the config file, reloaded the web page and landed on the the web page "Update required" with the "Start update now" button. Again I got stuck in the database schema check.

Next I tried to start the upgrade via `occ upgrade`. This produced the same effect, but the output was written to the console rather than to the web page... as expected.

**Comment by SpamReceiver** [comment]
I now have tried to update with `occ upgrade --skip-migration-test`. The upgrade now completed, but I had to turn off the maintenance mode afterwards in a separate step.

Reloading the web page takes me to the login prompt, but the login is now spinning and does not load the files browser (which is my welcome page as per standard installation).

**Comment by nickvergessen** [comment]
> Maybe this prevents the successful completion of the web login.

Surely it does ;)

> How do you measure the installation size? 

Number of users/files, size of DB and storage

**Comment by SpamReceiver** [comment]
> >   How do you measure the installation size?
> Number of users/files, size of DB and storage
Consider it to be "small": private use just for the family.
The only considerable portion is the link to the external directories on localhost.

**Comment by SpamReceiver** [comment]
> >  Maybe this prevents the successful completion of the web login.

> Surely it does ;)

I am not sure what will happen if I again kill mysql... Is there anything else I could do?

**Comment by ownclouders** [comment]
Hey, this issue has been closed because the label `needs info` is set and there were no updates for 14 days. Feel free to reopen this issue if you deem it appropriate.

(This is an automated comment from [GitMate.io](https://gitmate.io).

**Comment by SpamReceiver** [comment]
I would like to add the following information in case anybody stumbles over this entry:

The updater got stuck with every update since I raised this issue. I had to return to the command line and use the <code>--skip-migration-test</code> option every time.

I have updated on the "stable" channel up to the latest 9.x version. This problem has never been solved.

Now have moved to 10.x stable channel. The migration took quite a long time with several "repair steps". However, now update works without getting stuck and without <code>--skip-migration-test</code>.
